### PR TITLE
added usdt

### PR DESCRIPTION
**Pull request by thehashmap**

### CODE REVIEW

File: src/explanations/usdt.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+
+    Another controversy surrounding Tether is its lack of transparency and audits. Tether claims that its USDT stablecoin is backed 1:1 by US dollars held in reserve, but the company has not undergone a full audit to prove this. In 2018, the New York Attorney General's office filed a lawsuit against Tether and Bitfinex, alleging that the companies had engaged in a cover-up to hide the loss of $850 million in client and corporate funds. The case is ongoing.
```
Comment (RobinNagpal) [inline src/explanations/usdt.yaml:44]: We need to capture references for all these facts listed here

File: src/explanations/usdt.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    Tether has been publishing quarterly audit [reports](https://tether.to/en/transparency/#reports) since 2021 with the help of firms such as Moore, Mha Cayman and BDO. However, most of the companies feel that those are just attestation which don't give the full picture.
+
+
+    ### How They are Backed
+
+
+    USDT is meant to be backed 1:1 by the stable asset, meaning that for every USDT in circulation, Tether Limited is supposed to hold an equivalent value of the stable asset in reserve. According to Tether Limited, the stable asset backing USDT is typically held in the form of US Treasury bills, which are low-risk, short-term debt securities issued by the US government. However, as mentioned above, the company has not undergone a full audit to prove that it has the necessary reserves to back all of the USDT in circulation.
+
+
+    ### Issues and Hacks
```
Comment (RobinNagpal) [inline src/explanations/usdt.yaml:38]: The below three sections seems to be the same i.e. 
1) Transparency and Audits
2) Issues and Hacks
3) Good and Bad News 

and can be consolidated into one

File: src/explanations/usdt.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    ### Good and Bad News
+
+    Despite the controversies surrounding Tether Limited and Bitfinex, USDT has become one of the most widely used stablecoins in the cryptocurrency market. It is accepted on a wide range of exchanges and is often used as a "bridge currency" to facilitate trades between different cryptocurrencies. This widespread adoption has helped to make USDT a key player in the cryptocurrency market.
+
+
+    However, the lack of transparency and audits has also led to concerns about the stability of USDT and the potential for fraud or mismanagement. Until Tether Limited undergoes a full audit and proves that it has the necessary reserves to back all of the USDT in circulation, these concerns are likely to continue.
+
+
+    ### References
```
Comment (RobinNagpal) [inline src/explanations/usdt.yaml:61]: We should list down all the References

File: src/explanations/usdt.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+    Tether and Bitfinex have also faced security and liquidity issues. In August 2016, Bitfinex was hacked and lost approximately 120,000 bitcoins, worth over $70 million at the time. In November 2017, $31 million worth of USDT was stolen from the company. Tether has also struggled to meet withdrawal requests in the past and has not produced an audit showing that it has the necessary reserves to back all of the USDT in circulation.
+
+
+    In September 2022, Bitfinex and Tether (referred to in court records as B/T) were ordered by the New York District Court to produce documents showing the backing of USDT, the outcome of which is still pending. The court also ordered B/T to stop loaning out Tether's stablecoins and to provide a full accounting of the Tether reserve. You can read more about it [here](https://cointelegraph.com/news/new-york-judge-orders-tether-to-document-usdt-backing).
+
+
```
Comment (RobinNagpal) [inline src/explanations/usdt.yaml:52]: We should add another section, "growth and adoption", which should be on the positive side as USDT is the largest stable coin.

File: src/course.yaml [hunk 5]
```diff
@@ -24,8 +24,24 @@ topics:
       This chapter talks about USDC.
     explanations: usdc.yaml
     questions: usdc.yaml
-    status: In Progress
-    completionWeek: In Progress
+    status: Completed
+    completionWeek: Completed
+  - title: USDT
+    key: usdt
+    details: >
```
Comment (RobinNagpal) [inline src/course.yaml:31]: Add more info for details.

File: src/course.yaml [hunk 6]
```diff
@@ -24,8 +24,24 @@ topics:
...
+      This chapter talks about USDT.
+    explanations: usdt.yaml
+    questions: usdt.yaml
+    status: Completed
+    completionWeek: Completed
+  - title: DAI
+    key: dai
+    details: >
```
Comment (RobinNagpal) [inline src/course.yaml:39]: Add more info for details.

File: src/explanations/dai.yaml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+
+    MakerDAO and the DAI stablecoin are designed to be transparent and auditable. The smart contracts that govern the Collateralized Debt Positions (CDPs) and the minting and burning of DAI are open-source, which allows anyone to review the code and understand how the system works.
+
+
+    In addition, MakerDAO has undergone several independent audits by reputable firms, which have found that the system is secure and functioning as intended. In 2019, Callisto Network Security conducted a security audit of the Dai Token (DAI) smart contract which identified four issues, including three of low severity and one of owner privileges (which could potentially be risky for investors). No critical security issues were discovered. The audit specifically identified known vulnerabilities of ERC-20 tokens, the ability for the contract owner to block transferring, a missing event in ERC20 compliance, and the need to check input addresses for empty values. Overall, the audited smart contract was determined to be safe to deploy, with only low severity issues found during the audit. [Here](https://callisto.network/dai-token-dai-security-audit/) is the complete report.
```
Comment (RobinNagpal) [inline src/explanations/dai.yaml:38]: No Audits done post 2019?

File: src/explanations/dai.yaml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    The value of the collateral used to mint DAI must always exceed the value of the DAI itself, a process known as overcollaterization. This is to ensure that the collateral can cover the value of the DAI in the event that the market value of the collateral falls. If the value of the collateral falls below the value of the DAI, the vault account may be liquidated, with a ratio of 1.5 collateral to 1 DAI.
+
+
+    To ensure that undercollateralized accounts have sufficient funds to be liquidated, the MakerDAO system offers "keeper" accounts. Keepers are special user accounts that stake their cryptocurrency holdings to provide liquidity to pay off the collateral of vault accounts that lose value over time. This helps to maintain a steady supply of DAI in circulation to match the value of one US dollar.
+
+
+    ### Issues and Hacks
+
+
+    Like any complex system, MakerDAO and DAI have experienced some issues and challenges. In 2020, the value of Ethereum plummeted, which caused many CDPs to be liquidated and resulted in a significant amount of DAI being minted. This led to a temporary increase in the supply of DAI, which caused its value to drop below one US dollar.
```
Comment (RobinNagpal) [inline src/explanations/dai.yaml:56]: We should capture and include more specifics about this event
https://medium.com/@whiterabbit_hq/black-thursday-for-makerdao-8-32-million-was-liquidated-for-0-dai-36b83cac56b6

File: src/questions/usdt.yaml [hunk 9]
```diff
@@ -0,0 +1,170 @@
+- uuid: 0ba7d128-2865-4cf3-a81b-76823f99fade
+  type: SingleChoice
+  content: Who founded Tether Limited?
+  hint: Three founders
```
Comment (RobinNagpal) [inline src/questions/usdt.yaml:4]: Hit reveals the answer so we can use `NoHint` in this case

File: src/questions/usdt.yaml [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+    - content: JL van der Velde
+      key: B
+    - content: Craig Sellars
+      key: C
+    - content: Brock Pierce
+      key: D
+- uuid: 235c69d6-f26b-4872-9a43-1c67c6244bf7
+  type: SingleChoice
+  content: What is the main controversy surrounding Tether Limited and its affiliated exchange, Bitfinex?
+  hint: All the options seem correct.
```
Comment (RobinNagpal) [inline src/questions/usdt.yaml:89]: Hit reveals the answer so we can use `NoHint` in this case

File: src/questions/usdt.yaml [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+  content: How is USDT meant to be backed according to Tether Limited?
+  hint: NoHint
+  explanation: USDT is meant to be backed 1:1 by the stable asset, typically held in the form of US Treasury bills.
+  answerKeys:
+    - B
+  difficultyLevel: Low
+  choices:
+    - content: 1:1 by a basket of commodities such as gold and oil
+      key: A
+    - content: 1:1 by the stable asset, typically held in the form of US Treasury bills
```
Comment (RobinNagpal) [inline src/questions/usdt.yaml:114]: 1:1 by the liquid stable asset, like assets in the form is USD case or in the form of US Treasury bills

File: src/questions/usdt.yaml [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+  difficultyLevel: Low
+  choices:
+    - content: They were worried about the volatility of the stablecoin
+      key: A
+    - content: They believed that Tether Limited was engaging in market manipulation
+      key: B
+    - content: They were concerned about the security of the stablecoin
+      key: C
+    - content: They were concerned about the lack of know-your-customer (KYC) checks and other safeguards as the stablecoin circulated on various blockchains
+      key: D
```
Comment (RobinNagpal) [inline src/questions/usdt.yaml:170]: We should add a section about group and adoption and then couple of questions about it

File: src/questions/dai.yaml [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+    - A
+  difficultyLevel: Low
+  choices:
+    - content: Yes
+      key: A
+    - content: No
+      key: B
+- uuid: 6bacd29b-b431-48a6-82fb-50f31becf94f
+  type: SingleChoice
+  content: What did the security audit of the Dai Token (DAI) smart contract conclude about the smart contract's safety for deployment?
```
Comment (RobinNagpal) [inline src/questions/dai.yaml:50]: I guess there had been many audits.  Typically every release is audited and MakerDAO had a lot of code releases.

So we should be specific about certain audit. But that will be too specific

File: src/explanations/dai.yaml [hunk 14]
```diff
@@ -0,0 +1,82 @@
+- title: DAI coin
+  shortTitle: DAI
+  key: dai-introduction
+  details: |
+    ### About
```
Comment (RobinNagpal) [inline src/explanations/dai.yaml:5]: We should add a section related to `Maintaining Peg and Liquidation`

File: src/questions/dai.yaml [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+      key: A
+    - content: It had only low severity issues
+      key: B
+    - content: It had one critical security issue
+      key: C
+    - content: It had two low severity issues and one critical security issue
+      key: D
+- uuid: eccb11dd-0c00-45ec-8a0d-76a3c2b0bbdf
+  type: SingleChoice
+  content: What is the Dai Savings Rate (DSR)?
```
Comment (RobinNagpal) [inline src/questions/dai.yaml:67]: Is it for saving or for earning interest on DAI?

File: src/explanations/dai.yaml [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+    DAI is used in various ways, including as a medium of exchange, a store of value, and a collateral asset. It has gained widespread adoption in the cryptocurrency community and has been used in a number of decentralized finance (DeFi) applications.
+
+
+    ### Minting and Burning DAI
+
+
+    Dai is generated when users deposit collateral assets into [Maker Vaults](https://makerdao.com/en/whitepaper/#maker-vaults) (Maker Vaults facilitate the generation of Dai against locked-up collateral. Users open Vaults via the Oasis Borrow dashboard or one of many other user interfaces that allow the management of Vault operations ) within the Maker Protocol. This process allows users to access liquidity and enters Dai into circulation. Others can obtain Dai by purchasing it from brokers or exchanges, or by receiving it as payment.
+
+
+    Once generated, purchased, or received, Dai can be used in the same way as other cryptocurrencies. It can be sent to others, used as payment for goods and services, and even saved through a feature of the Maker Protocol called the [Dai Savings Rate (DSR)](https://ethereumprice.org/guides/article/dai-savings-rate-explained/).
```
Comment (RobinNagpal) [inline src/explanations/dai.yaml:23]: Is it for saving or for earning interest on DAI?

File: src/explanations/dai.yaml [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+
+
+    If a user who has borrowed DAI against collateral decides to pay back the loan, the DAI coins are "burned" and the collateral deposit in the account is available for withdrawal. The process of minting and burning DAI helps to maintain its value at or near one US dollar.
+
+
+    The MCD_DAI_JOIN adapter contract in the Maker Protocol is authorized to call the mint and burn functions of the DAI token in order to increase or decrease the total supply of DAI. These functions are also used when users deposit or withdraw DAI from the Dai Savings Rate contract.
```
Comment (RobinNagpal) [inline src/explanations/dai.yaml:29]: `MCD_DAI_JOIN` IS this the primary code that handles it? We should also include the reference to this

File: src/explanations/dai.yaml [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+
+
+    [Recently](https://cointelegraph.com/news/makerdao-should-seriously-consider-depegging-dai-from-usd-founder), MakerDAO founder Rune Christensen has suggested that the organization should "seriously consider" depegging its stablecoin, DAI, from the US dollar following recent sanctions on cryptocurrency mixer Tornado Cash and the freezing of USDC addresses by Circle. Christensen raised concerns over DAI's heavy reliance on a centralized asset, USDC, which represents over 50% of MakerDAO's DAI collateralization, according to Dai Stats. The stablecoin is currently the fourth largest USD-pegged stablecoin in the cryptocurrency market, with a market cap of $7 billion. The idea of converting all USDC from MakerDAO's peg stability module into ether was met with criticism, with Ethereum co-founder Vitalik Buterin calling it a "risky and terrible idea."
+
+
+    Despite these concerns, DAI has proven to be a reliable and stable stablecoin, and its use is likely to continue to grow in the future.
+
```
Comment (RobinNagpal) [inline src/explanations/dai.yaml:75]: We should also add in a couple of sections that the data related to DAI backing is available onchain and can be queried by anyone. Also there are a number of graphing application that show the runtime information of the system.

File: src/explanations/dai.yaml (hunk 7 above)
Comment (thehashmap) [reply src/explanations/dai.yaml:38]: I could not find any audit reports other than the one by Callisto Network

**Pull request merged**